### PR TITLE
Use docker env file as source of Fn server configuration opts

### PR DESCRIPTION
**Pull request by denismakogon**
Closes: #122

### CODE REVIEW

File: start.go [hunk 1]
```diff
@@ -17,25 +17,19 @@ func startCmd() cli.Command {
 		Usage:  "start a functions server",
 		Action: start,
 		Flags: []cli.Flag{
-			cli.StringFlag{
```
Comment (treeder) [inline start.go]: Why did you remove this?

File: start.go [hunk 2]
```diff
@@ -44,10 +38,10 @@ func start(c *cli.Context) error {
 		"--name", "fnserver",
 		"-v", fmt.Sprintf("%s/data:/app/data", wd),
 		"-v", "/var/run/docker.sock:/var/run/docker.sock",
-		"-p", "8080:8080",
```
Comment (treeder) [inline start.go]: Why change this?

File: start.go (hunk 1 above)
Comment (denismakogon) [reply start.go]: because it's not necessary anymore, you can put one to env file or just set FN_LOG_LEVEL to DEBUG and you'll be fine.

File: start.go (hunk 2 above)
Comment (denismakogon) [reply start.go]: because it can be really whatever it can, you have FN_PORT which would change port, so it's not always 8080.

File: start.go (hunk 1 above)
Comment (treeder) [reply start.go]: How would FN_LOG_LEVEL make it into the container?

File: start.go (hunk 2 above)
Comment (treeder) [reply start.go]: I'd much prefer this: https://github.com/fnproject/cli/pull/79 vs what you have here.

File: start.go (hunk 1 above)
Comment (denismakogon) [reply start.go]: through env file, that's the purpose of this whole patch - let user pass fn server opts to the container.

Comment (treeder) [reply start.go]: Ok, well please don't make .env file required, you're making a lot of assumptions here.

File: start.go [hunk 3]
```diff
@@ -17,25 +17,19 @@ func startCmd() cli.Command {
...
-				Name:  "log-level",
-				Usage: "--log-level DEBUG to enable debugging",
-			},
 			cli.BoolFlag{
 				Name:  "detach, d",
 				Usage: "Run container in background.",
 			},
+			cli.StringFlag{
+				Name: "config, c",
+				Usage: "Absolute path to Fn server configuration options file.",
```
Comment (treeder) [inline start.go]: And why not call it `--env-file` like docker? Also, does it really require an absolute path?

File: start.go (hunk 1 above)
Comment (denismakogon) [reply start.go]: it's not required by default

File: start.go (hunk 2 above)
Comment (denismakogon) [reply start.go]: okay, will revert.

File: start.go (hunk 1 above)
Comment (treeder) [reply start.go]: I'm saying, please keep log-level there. I don't want to have to make an env file to get debugging output.

File: start.go (hunk 3 above)
Comment (denismakogon) [reply start.go]: because we're not trying to reimplement docker CLI here, moreover, `--env-file` name is not verbose or helpful. The name of the opt should be very clear, that's why it was named as `--config`

Comment (denismakogon) [reply start.go]: well, path doesn't suppose to be absolute, but it's more clear than even docker CLI helper string: `Read in a file of environment variables`

Comment (treeder) [reply start.go]: But this isn't really a config file, it's a file containing env vars. If we ever support a config file, it probably (almost surely) wouldn't be in this format. This file format dates back to the Ruby days: https://github.com/bkeepers/dotenv

Also, the the flags should match docker run (like -d) since this is just a convenience command that passes right through to docker run.

**Comment by denismakogon** [comment]
@treeder so, can we merge this?

**Comment by treeder** [comment]
If you change it to --env-file to match docker, then yes.

**Comment by denismakogon** [comment]
@treeder done.
